lib/datadog/core/configuration/settings.rb

```diff
@@ -107,6 +107,19 @@ def initialize(*_)
           o.env Core::Environment::Ext::ENV_API_KEY
         end
 
+        # Datadog Application key.
+        #
+        # For internal use only.
+        # NOTE: This should be only used when interacting with the Datadog API directly.
+        # TODO: mark this as experimental?
+        #
+        # @default `DD_APP_KEY` environment variable, otherwise `nil`
+        # @return [String, nil]
+        option :app_key do |o|
+          o.type :string, nilable: true
+          o.env Core::Environment::Ext::ENV_APP_KEY
+        end
```

Comment:
+1 yes please I think this should be moved inside the api guard group and marked as experimental, as it's not clear at this point this is going to stay